src/components/search-wizard/Wizard.jsx

```diff
@@ -34,33 +35,52 @@ export default function SearchWizard() {
     }
   }, [queryLayersConfig]);
 
-  if (queryLayersConfig.status === 'loading') {
+  if (queryLayersConfig.status === 'loading' || !state.context.searchLayers) {
     return null;
   }
 
   return (
-    <div className="p-2">
-      {state.matches('queryLayers') ? (
-        <SelectMapData
-          queryLayers={queryLayers}
-          machineState={state}
-          machineSend={send}
+    <div className="flex h-full flex-col">
+      {state.matches('selectLayers') ? (
+        <SelectMapData queryLayers={queryLayers} />
+      ) : null}
+      {state.matches('searching') || state.matches('result') ? (
+        <Progress
+          searchLayers={state.context.searchLayers}
+          results={state.context.resultLayers}
         />
       ) : null}
       {state.matches('advanced') ? <AdvancedFilter /> : null}
-      Selected query layer ids:{' '}
-      {JSON.stringify(state.context.queryLayers.map((config) => config.id))}
+      {state.matches('error') ? (
+        <p>{JSON.stringify(state.context.error, null, 2)}</p>
+      ) : null}
+      <div className="justify-self-end border-t border-t-slate-300 p-2">
+        <Button
+          appearance={Button.Appearances.solid}
+          color={Button.Colors.primary}
+          className="w-full"
+          size={Button.Sizes.xl}
+          busy={state.matches('searching')}
+          disabled={state.context.searchLayers.length === 0}
+          onClick={() => send('SEARCH')}
+        >
+          Search{' '}
+          {state.context.searchLayers.length
+            ? `${state.context.searchLayers.length} Layer${
+                state.context.searchLayers.length > 1 ? 's' : ''
+              }`
```

Comment:
**suggestion**: consider something like https://www.npmjs.com/package/pluralize if this becomes common.